src/ng/directive/ngOptions.js

```diff
@@ -325,12 +325,28 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         // The option values were already computed in the `getWatchables` fn,
         // which must have been called to trigger `getOptions`
         var optionValues = valuesFn(scope) || [];
+        var optionValuesKeys;
 
-        var keys = Object.keys(optionValues);
-        keys.forEach(function getOption(key) {
+
+        if (!keyName && isArrayLike(optionValues)) {
```

Comment:
OK, I hear that you are saying that the idea of iterating over an object's keys is a bad idea.
Unfortunately this would be a momentous breaking change for many apps and so we will not be changing this in the near future - if at all in 1.x.

Other than that are you happy with the PR?